tests/manage/nfs_feature/test_nfs_feature_negative_testcases.py

```diff
@@ -0,0 +1,519 @@
+
+
+log = logging.getLogger(__name__)
+# Error message to look in a command output
+ERRMSG = "Error in command"
+NAMESPACE = "openshift-storage"
+STORAGE_CLUSTER_OBJ = ocp.OCP(kind="Storagecluster", namespace=NAMESPACE)
+CONFIG_MAP_OBJ = ocp.OCP(kind="Configmap", namespace=NAMESPACE)
+POD_OBJ = ocp.OCP(kind="Pod", namespace=NAMESPACE)
+SERVICE_OBJ = ocp.OCP(kind="Service", namespace=NAMESPACE)
+PVC_OBJ = ocp.OCP(kind=constants.PVC, namespace=NAMESPACE)
+PV_OBJ = ocp.OCP(kind=constants.PV, namespace=NAMESPACE)
+NFS_SC = "ocs-storagecluster-ceph-nfs"
+SC = ocs.OCS(kind=constants.STORAGECLASS, metadata={"name": NFS_SC})
+TEST_FOLDER = "test_nfs"
+
+
+@tier1
+@skipif_ocs_version("<4.10")
+@skipif_ocp_version("<4.10")
+@skipif_managed_service
+@skipif_disconnected_cluster
+@skipif_proxy_cluster
+class TestNfsUnavailableInLowerODFversions(ManageTest):
+    """
+    Test nfs feature is not available for ODF cluster version below 4.11
+
+    """
+
+    def test_nfs_not_available_before_odf_4_11(self):
+        """
+        This test is to validate nfs feature is not available for ODF cluster version below 4.11
+
+        Steps:
+        1:- Check ocs version < 4.11
+        2:- Checks cephnfs resources not available by default
+        3:- Enable nfs feature for storage-cluster using patch command
+        4:- Checks cephnfs resources not available
+        5:- Upgrade OCS to >=4.11
+        6:- Enable nfs feature
+        7:- Check cephnfs resource running
+        8:- Disable nfs feature
+
+        """
+        ocs_version = version.get_ocs_version_from_csv(
+            only_major_minor=False, ignore_pre_release=True
+        )
+        log.info(f"odf version---- {ocs_version}")
+        if ocs_version < version.VERSION_4_11:
+            log.info("ODF version is <4.11")
+        else:
+            log.warning("ODF version is not <4.11")
+
+        # Checks cephnfs resources not available by default
+        cephnfs_resource = STORAGE_CLUSTER_OBJ.exec_oc_cmd("get cephnfs")
+        if cephnfs_resource is None:
+            log.info(f"No resources found in openshift-storage namespace.")
+        else:
+            log.error("nfs feature is enabled by default")
+
+        nfs_spec_enable = '{"spec": {"nfs":{"enable": true}}}'
+
+        # Enable nfs feature for storage-cluster using patch command
+        assert STORAGE_CLUSTER_OBJ.patch(
+            resource_name="ocs-storagecluster",
+            params=nfs_spec_enable,
+            format_type="merge",
+        ), "storagecluster.ocs.openshift.io/ocs-storagecluster not patched"
+
+        # Checks cephnfs resources not available
+        cephnfs_resource = STORAGE_CLUSTER_OBJ.exec_oc_cmd("get cephnfs")
+        if cephnfs_resource is None:
+            log.info(f"No resources found in openshift-storage namespace.")
+        else:
+            log.error(f"nfs feature is enabled for ODF cluster version, {ocs_version}")
+
+        # Upgrade OCS
+        run_ocs_upgrade()
```

Comment:
OCS upgrade is done with explicit jobs and is not supposed to be part of any test but `test_ocs_upgrade`. 
I suggest that for validating this flow, you will add a pre-upgrade test case which will validate NFS is not enabled on clusters with versions lower than 4.11 and a post upgrade test which will do the rest of the test case. 
You can use this test file for reference - https://github.com/red-hat-storage/ocs-ci/blob/master/tests/ecosystem/upgrade/test_resources.py